### PR TITLE
feat: add cameraMode options for tracking location changes

### PR DESCRIPTION
**Pull request by torrybr**
Allow users to set the `CameraMode` for tracking location changes. CameraMode options and sdk information can be found [here](https://docs.mapbox.com/android/legacy/maps/guides/location-component/#cameramode).

Closes https://github.com/ramani-maps/ramani-maps/issues/72

### CODE REVIEW

**Comment by torrybr** [comment]
Whats the best approach for listening to `CameraTrackingChanges`?  Something such as:

```
    this.locationComponent.addOnCameraTrackingChangedListener(object : OnCameraTrackingChangedListener {
        override fun onCameraTrackingDismissed() {
            // Optionally handle if the tracking is dismissed
        }

        override fun onCameraTrackingChanged(currentMode: Int) {
            // When the camera mode changes, ensure it is set to the desired mode
            if (currentMode != cameraMode) {
                locationComponent.cameraMode = cameraMode
            }
        }
    })
```

File: ramani-maplibre/src/main/java/org/ramani/compose/MapLibre.kt [hunk 1]
```diff
@@ -144,6 +147,7 @@ fun MapLibre(
                 currentLocationStyling,
                 userLocation,
                 renderMode,
+                cameraMode
```
Comment (JonasVautherin) [inline ramani-maplibre/src/main/java/org/ramani/compose/MapLibre.kt:150]: Nitpick :innocent: 

```suggestion
                cameraMode,
```

File: ramani-maplibre/src/main/java/org/ramani/compose/MapLibre.kt [hunk 2]
```diff
@@ -226,7 +230,8 @@ private fun MapLibreMap.setupLocation(
     locationRequestProperties: LocationRequestProperties?,
     locationStyling: LocationStyling,
     userLocation: MutableState<Location>?,
-    renderMode: Int
+    renderMode: Int,
+    cameraMode: Int
```
Comment (JonasVautherin) [inline ramani-maplibre/src/main/java/org/ramani/compose/MapLibre.kt:234]: Nitpick again :see_no_evil: 
```suggestion
    cameraMode: Int,
```

**Review by JonasVautherin — CHANGES_REQUESTED** [review]
Not sure why I took so long to review that, sorry (I had a couple busy weeks).

Would you mind rebasing on `main` (and I took this opportunity to add two nitpicks :innocent:) and then we can merge it?

Thanks a lot! :rocket:

**Comment by torrybr** [comment]
Rebased and comments addressed 👍  Before we merge it, Im wondering If I need the `addOnCameraTrackingChangedListener` as in the linked examples because simply adding  `this.locationComponent.cameraMode = cameraMode` is not triggering the cameraMode change. Would you mind taking a quick look to confirm? Thanks for all your help!

**Comment by JonasVautherin** [comment]
> this.locationComponent.cameraMode = cameraMode is not triggering the cameraMode change

Oh, so it's not working then?

Which examples are you referring to? Is there a small application that I could quickly run to see what's happening?

**Comment by torrybr** [comment]
I updated the [`Location` example](examples/location/app/src/main/java/org/ramani/example/location/MainActivity.kt) so that it uses a tracking camera mode. You should be able to set a gps route through android studio to test! Thanks!

**Comment by JonasVautherin** [comment]
I haven't had time to try it yet, but it's still planned! In the meantime, did you manage to make it work?

**Comment by torrybr** [comment]
I updated with a `addOnCameraTrackingChangedListener` which seems to work properly and rotate the map as expect when loading the `Location` example i mentioned above. Im curious to see what you think! Im not quite sure why it requires this listener to function as expected

**Comment by JonasVautherin** [comment]
Amazing, I'll test that ASAP!

**Comment by torrybr** [comment]
Thanks for taking a look!

**Review by JonasVautherin — APPROVED** [review]
Just tested it and it works! Thanks a lot! :rocket:

**Pull request merged**